### PR TITLE
feat: allow configuring pypi insecure host

### LINKED ISSUE #1982: Invalid peer certificate with `--tls-no-verify`
### Checks

- [X] I have checked that this issue has not already been reported.

- [X] I have confirmed this bug exists on the [latest version](https://github.com/prefix-dev/pixi/releases) of pixi, using `pixi --version`.

### Reproducible example

image

### Issue description

I installed a pypi package published on nexus using the latest pixi on the company's intranet server, but encountered `invalid peer certificate: UnknownIssuer`
image

### Expected behavior

`--tls-no-verify` should work

### PR DESCRIPTION
**Pull request by zen-xu**
close #1982

### CODE REVIEW

**Comment by ruben-arts** [comment]
This looks good to me, @tdejager could you review this?

**Comment by tdejager** [comment]
Implementation looks fine, the question for me would be if we want `uv_configuration` as a dependency because then we can never publish this specific crate, otherwise we need to define the type and conversion ourselves. Personally, I'd be for that.
cc @ruben-arts @baszalmstra

File: crates/pixi_config/Cargo.toml [hunk 1]
```diff
@@ -26,6 +26,7 @@ serde_json = { workspace = true }
 toml_edit = { workspace = true, features = ["serde"] }
 tracing = { workspace = true }
 url = { workspace = true }
+uv-configuration = { workspace = true }
```
Comment (tdejager) [inline crates/pixi_config/Cargo.toml:29]:
```suggestion
uv-configuration = { workspace = true }
```

Could we redefine the types here and make a conversion method in `pixi_uv_conversions` its better to keep uv dependencies out, as this prohibits us publishing the crates.

**Review by tdejager — CHANGES_REQUESTED** [review]

File: crates/pixi_config/Cargo.toml (hunk 1 above)
Comment (zen-xu) [reply crates/pixi_config/Cargo.toml]: No problem

**Review by tdejager — APPROVED** [review]
Thanks looks great!

**Pull request merged**